contracts/src/v0.1/RequestResponseConsumerBase.sol

```diff
@@ -22,16 +22,71 @@ abstract contract RequestResponseConsumerBase {
      * @return req request in memory
      */
     function buildRequest(bytes32 jobId) internal view returns (Orakl.Request memory req) {
-        return req.initialize(jobId, address(COORDINATOR), COORDINATOR.fulfillDataRequest.selector);
+        return
+            req.initialize(
+                jobId,
+                address(COORDINATOR),
+                COORDINATOR.fulfillDataRequestUint256.selector
+            );
     }
 
-    function fulfillDataRequest(uint256 requestId, uint256 response) internal virtual;
+    function fulfillDataRequestUint256(uint256 requestId, uint256 response) internal virtual;
```

Comment:
I wanted to suggest to keep function names, event names and others all same. It creates a unique function selector there is at least one parameter type different. However, it is not possible to access overloaded function through function selector in Solidity 😢 https://github.com/ethereum/solidity/issues/3556

Reply:
Do you mean we can just keep different function names for different data types?

Reply:
We can and it would be nicer, however, we need to have an access to function selector from inside of smart contract and that is currently not supported in Solidity.